src/pnm/header.rs

```diff
@@ -11,8 +11,14 @@ pub enum SampleEncoding {
 }
 
 /// Denotes the category of the magic number
+///
+/// DEPRECATED: The name of this enum will be changed to [`PnmSubtype`].
```

Comment:
I opted to use a comment to say that this was deprecated as it does not require any `#[allow(deprecated)]` tags in the rest of the code.